cmd/agent/app/agent.go

```diff
@@ -104,6 +106,18 @@ func init() {
 
 func run(ctx context.Context, karmadaConfig karmadactl.KarmadaConfig, opts *options.Options) error {
 	klog.Infof("karmada-agent version: %s", version.Get())
+
+	if opts.ProfileOpts.EnableProfile {
+		addr := fmt.Sprintf(":%d", opts.ProfileOpts.ProfilePort)
+		klog.Infof("Starting profiling on port %s", opts.ProfileOpts.ProfilePort)
+		go func() {
+			if err := http.ListenAndServe(addr, nil); err != nil {
```

Comment:
1. we shouldn't use default `DefaultServeMux`
2. we can abstract a common function to start the server in `pkg/sharedcli/profileflag/`, like:
```
func ListenAndServe(opts Options)
```